kobo/apps/superuser_stats/tasks.py

```diff
@@ -79,6 +87,235 @@ def get_row_for_country(code_: str, label_: str):
+            twelve_submission_count['submissions_count'],
+        ])
+
+    headers = [
+        'Username',
+        'Date Joined',
+        'Assets 3m',
+        'Assets 6m',
+        'Assets 12m',
+        'Submissions 3m',
+        'Submissions 6m',
+        'Submissions 12M',
+    ]
+
+    default_storage = get_storage_class()()
+    with default_storage.open(output_filename, 'wb') as output:
+        writer = unicodecsv.writer(output)
+        writer.writerow(headers)
+        writer.writerows(data)
+
+
+@shared_task
+def generate_domain_report(output_filename: str, start_date: str, end_date: str):
+    emails = User.objects.filter(
+        date_joined__range=(start_date, end_date),
+    ).values_list('email', flat=True)
+
+    # get a list of the domains
+    domains = [email.split('@')[1] if '@' in email else '!!invalid: ' + email
+               for email in emails
+               ]
+    domain_users = Counter(domains)
+
+    # get a count of the assets
+    domain_assets = {
+        domain:
+            Asset.objects.filter(
+                owner__email__endswith='@' + domain,
+                date_created__range=(start_date, end_date),
+            ).count()
+            for domain in domain_users.keys()
+    }
+
+    # get a count of the submissions
+    domain_submissions = {
+        domain: ReadOnlyKobocatMonthlyXFormSubmissionCounter.objects.annotate(
+            date=Cast(
+                Concat(F('year'), Value('-'), F('month'), Value('-'), 1),
+                DateField(),
+            )
+        ).filter(
+            user__email__endswith='@' + domain,
+            date__range=(start_date, end_date),
+        ).aggregate(
+            Sum('counter')
+        )['counter__sum']
+        if domain_assets[domain] else 0
+        for domain in domain_users.keys()
+    }
+
+    # create the CSV file
+    columns = ['Email Domain', 'Users', 'Projects', 'Submissions']
+
+    default_storage = get_storage_class()()
+    with default_storage.open(output_filename, 'wb') as output:
+        writer = unicodecsv.writer(output)
+        writer.writerow(columns)
+
+        for domain, users in domain_users.most_common():
+            row = [
+                domain,
+                users,
+                domain_assets[domain],
+                domain_submissions[domain]
+            ]
+            writer.writerow(row)
+
+
+@shared_task
+def generate_forms_count_by_submission_range(output_filename):
+    # List of submissions count ranges
+    ranges = [
+        (0, 0),
+        (1, 500),
+        (501, 1000),
+        (1001, 10000),
+        (10001, 50000),
+        (50001, None)
+    ]
+
+    # store data for csv
+    data = []
+
+    today = datetime.datetime.today()
+    date = today - relativedelta(months=12)
+
+    for r in ranges:
+        forms_count = KobocatXForm.objects.filter(
+            date_created__gte=date,
+            num_of_submissions__range=r,
+        ).values('pk').count()
```

Comment:
When using `count()` , IIRC, narrowing down the number of fields does not really matter on performance (i.e. `values('pk')`) . 
Please have a look and let me know if I'm wrong. 
If it's not needed, we can safely remove it.